### ISSUE TITLE
Zooming in High Res PDFs blurs text and making them unreadable

### ISSUE THREAD

**Issue by CodingFabian** [mozilla/pdf.js#5072]
￼When zooming in on some large drawing/floorplan/map pdfs, the zoomed text becomes unreadable.
It looks like it is rendered to the canvas only once and then upscaled.
As an example see this map:
https://public.centerdevice.de/8ddc0fe8-ecc6-456a-bdac-4e5f58b22f35

This is rendered at 400% using acrobat:
![2014-07-23_1503](https://cloud.githubusercontent.com/assets/176132/3673255/e2654d6c-126a-11e4-869a-723de2c1bb60.png)

and this is how it looks in pdfjs:
![2014-07-23_1502](https://cloud.githubusercontent.com/assets/176132/3673257/e73ce660-126a-11e4-96bd-8d05d6a1dde7.png)

This is especially a problem on large maps which require much zooming to be actually readable. if the initial drawing was too small the zooming doesnt help and only enlarges garbage

**Comment by tadashi-koya** [comment]
Deare team.

I am also troubled by the same event.

I attempt #4834 case about  maxCanvasPixels tunning but Nothing change.

Is there any tips/workaround for this problem?

regards.

**Comment by timvandermeij** [comment]
The only good solution is the one that @yurydelendik provided above, but someone needs to implement that, so there is no solution at the moment.

**Comment by tadashi-koya** [comment]
Thank you for your advice.

I understand that there is no implementation now.

regards.

**Comment by yurydelendik** [comment]
Blocked by #6419

**Comment by Snuffleupagus** [comment]
Duplicate of #6419
